.circleci/config.yml

```diff
@@ -28,9 +28,9 @@ store-log-artifacts: &store-log-artifacts
     path: log
     destination: log
 
-npm-cache-key: &npm-cache-key node-v10-{{ checksum "yarn.lock" }}-5
-bundle-cache-key: &bundle-cache-key v2-bundler-gems-{{ .Environment.DB }}-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
-assets-cache-key: &assets-cache-key v1-asset-cache-{{ checksum "tmp/assets_related_checksums" }}
+npm-cache-key: &npm-cache-key porta-{{ .Environment.CACHE_VERSION }}-node-v10-{{ checksum "yarn.lock" }}
+bundle-cache-key: &bundle-cache-key porta-{{ .Environment.CACHE_VERSION }}-bundler-gems-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}-{{ checksum "/tmp/db" }}
```

Comment:
What's the `/tmp/db`?...

Reply:
It's a hack. The bundle must be different for each DB, because oracle has it's own gems. Then the cache key must be different depending on the DB. This was implemented by calling `{{ .Environment.DB }}` But that never worked because `.Environment` only knows of variables provided by CircleCI. From the [reference](https://circleci.com/docs/configuration-reference/):

> {{ .Environment.variableName }} | The environment variable variableName (supports any environment variable  exported by CircleCI or added to a specific  context—not any arbitrary environment variable).

I found this way to use arbitrary environment variables from cache keys:

https://discuss.circleci.com/t/setting-environment-variable-from-script-command-for-later-use-in-cache-name/24036

I our case we are temporary storing the variable value in a file so its checksum can be part of the cache key

Different DB -> Different file content -> Different checksum -> Different cache key

Reply:
Thanks for the clarification! Yeah, I later found that you've been using the temporary file for the DB name. But... you also added `DB: <value>` to the config. Is this not available through `{{ .Environment.DB }}`? How would we set those vars?

Reply:
> Is this not available through {{ .Environment.DB }}?

No, only variables we manually add to CircleCI contexts and projects from the website. Nothing set inside pipelines is visible to `.Environment`.

The `DB: <value>` I added are needed to make the `$DB` variable available at `install-gem-dependencies`

When the executor is either `builder-with-mysql-ruby26`, `builder-with-postgres-ruby26` or `builder-with-oracle-ruby26`; `$DB` is available through `build-envs-postgresql`, `build-envs-postgresql` or `build-envs-postgresql`, where I added the `DB: <value>`.

When the executor is `builder-ruby26`. `$DB` is set to a given parameter.

See declaration: https://github.com/3scale/porta/blob/d24c38983618f3d3cea7becfb07178cd730d36e6/.circleci/config.yml#L348-L352

And call: https://github.com/3scale/porta/blob/d24c38983618f3d3cea7becfb07178cd730d36e6/.circleci/config.yml#L436-L438